src/Simplify.elm

```diff
@@ -5127,6 +5137,204 @@ subAndCmdBatchChecks moduleName { lookupTable, parentRange, fnRange, firstArg }
 
 
 
+-- HTML.ATTRIBUTES
+
+
+htmlAttributesClassListFalseElementError : { message : String, details : List String }
+htmlAttributesClassListFalseElementError =
+    { message = "In a Html.Attributes.classList, a tuple paired with False can be removed"
+    , details = [ "You can remove the tuple list element where the second part is False." ]
+    }
+
+
+htmlAttributesClassListChecks : CheckInfo -> List (Error {})
+htmlAttributesClassListChecks checkInfo =
+    let
+        listArg : Node Expression
+        listArg =
+            checkInfo.firstArg
+
+        getTupleWithSecond : Bool -> Node Expression -> Maybe { first : Node Expression, range : Range }
+        getTupleWithSecond bool expression =
+            case Node.value expression of
+                Expression.TupledExpression (first :: second :: []) ->
+                    if getBool bool checkInfo.lookupTable second then
+                        Just { range = Node.range expression, first = first }
+
+                    else
+                        Nothing
+
+                _ ->
+                    Nothing
+
+        singleElementListChecks : { a | element : Node Expression } -> List (Error {})
+        singleElementListChecks single =
+            case getTupleWithSecond True single.element of
+                Just tuple ->
+                    singleTrueChecks tuple
+
+                Nothing ->
+                    case getTupleWithSecond False single.element of
+                        Just _ ->
+                            [ Rule.errorWithFix htmlAttributesClassListFalseElementError
+                                checkInfo.fnRange
+                                [ Fix.replaceRangeBy (Node.range listArg) listCollection.emptyAsString ]
+                            ]
+
+                        Nothing ->
+                            []
+
+        singleTrueChecks : { a | first : Node Expression } -> List (Error {})
+        singleTrueChecks { first } =
```

Comment:
I'm quite sad we can only do this check if it's there is only a single element in a lot of cases (because we don't have a function that takes a list of classes without conditions). But I'm thinking that if there are only literal strings in the first value, then we can simplify the expression to use `Html.Attributes.class`:

```elm
Html.Attributes.classList
  [ ( "some-class", True )
  , ( """other-class""", True )
  , ( "third-class", True )
  ]
--> Html.Attributes.class "some-class other-class third-class"
```
I think this is actually more common in old codebases, because I have seen classList used when only a single item was actually conditional.

This can be done in a separate PR. Actually, probably better done as a separate PR :smile:

Reply:
ok next PR